src/cinder/gl/GlslProg.cpp

```diff
@@ -697,10 +718,20 @@ void GlslProg::loadShader( const string &shaderSource, const fs::path &shaderPat
 			}
 			log += "\n" + ss.str();
 		}
-#endif		
+#endif
+		// Since the GlslProg destructor will not be called after we throw, we must delete all
+		// owned GL objects here to avoid leaking. Any other attached shaders will be cleaned up
+		// when the program is deleted
+		glDeleteShader( handle );
+		glDeleteProgram( mHandle );
+
 		throw GlslProgCompileExc( log, shaderType );
 	}
+
 	glAttachShader( mHandle, handle );
+	glDeleteShader( handle );
```

Comment:
While I'm reading from the [docs](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glDeleteShader.xhtml) that this `glDeleteShader` is essentially flagging the shader for deletion once it is detached, I think this is confusing here (had to go look it up :) ).  Would there be any harm in moving these calls to lines 564-583 above, directly after the calls to `glDetatchShader`? This would mean you need one call to `glDeleteShader` per stage, but at least it would be clear what is going on.

If the extra calls seems ugly, I'd suggest adding a new helper function that checks if the shader handle is non-zero and if so, calls both `glDetachShader` followed by `glDeleteShader`.